ShowBuildMenu.sh

```diff
@@ -0,0 +1,261 @@
+	read CFGNAME
+	if [ $CFGNAME = ""]
+	then
+		CFGNAME="$CONFIG_NAME-$TEST_PLATFORM"
+		echo $CFGNAME
+	fi
+
+	mkdir -p $AVAILABLE_CONFIGURATIONS/$CFGNAME
+
+	if [ ! "$LIB_FILES" ]
+	then
+		cp $LIB_FILES $AVAILABLE_CONFIGURATIONS/$CFGNAME
+
+		if [ ! "$LIB_FILES2" ]
+		then
+			cp $LIB_FILES2 $AVAILABLE_CONFIGURATIONS/$CFGNAME
+		fi
+	fi
+
+	cp "src/NHibernate.Config.Templates/$CONFIG_NAME.cfg.xml" "$AVAILABLE_CONFIGURATIONS/$CFGNAME/hibernate.cfg.xml"
+	echo "Done setting up files. Please edit the connection string in file:"
+	echo "$AVAILABLE_CONFIGURATIONS/$CFGNAME/hibernate.cfg.xml"
+	echo "When you're done, don't forget to activate the configuration through the menu."
+	mainMenu
+}
+
+testSetupSqlServer() {
+	CONFIG_NAME="MSSQL"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupFirebird() {
+	CONFIG_NAME="FireBird"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupSqlite() {
+	CONFIG_NAME="SQLite"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""e
+	testSetupGeneric
+}
+
+testSetupPostgresql() {
+	CONFIG_NAME="PostgreSQL"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupMysql() {
+	CONFIG_NAME="MySql"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupMenu() {
+	echo "A. Add a test configuration for SQL Server."
+	echo "B. Add a test configuration for Firebird."
+	echo "C. Add a test configuration for SQLite."
+	echo "D. Add a test configuration for PostgreSQL."
+	echo "E. Add a test configuration for MySql."
+	echo "."
+	echo "X.  Exit to main menu."
+	echo "."
+
+	$BUILD_TOOL prompt ABCDEX
+
+	OPTION=$?
+	if	[ $OPTION -eq 5 ]
+	then
+		echo "Main menu"
+		mainMenu
+	elif [ $OPTION -eq 4 ]
+	then
+		echo "MySQL"
+		testSetupMysql
+		mainMenu
+	elif [ $OPTION -eq 3 ]
+	then
+		echo "PostgreSQL"
+		testSetupPostgresql
+		mainMenu
+	elif [ $OPTION -eq 2 ]
+	then
+		echo "Sqlite"
+		testSetupSqlite
+		mainMenu
+	elif [ $OPTION -eq 1 ]
+	then
+		echo "Firebird"
+		testSetupFirebird
+		mainMenu
+	elif [ $OPTION -eq 0 ]
+	then
+		echo "SQL Server"
+		testSetupSqlServer
+		mainMenu
+	fi
+}
+
+testRun(){
+	eval "dotnet test ./src/NHibernate.Test/NHibernate.Test.csproj" -f netcoreapp2.0
+	eval "dotnet test ./src/NHibernate.Test.VisualBasic/NHibernate.Test.VisualBasic.vbproj" -f netcoreapp2.0
+	mainMenu
+}
+
+generateAsync(){
+	dotnet msbuild /t:Restore ./src/NHibernate.sln
+
+	getAsyncGeneratorPath
+	cd src
+	eval "mono ../$async_generator_path"
+	cd ..
+
+	mainMenu
+}
+
+getAsyncGeneratorPath(){
+	if [ "$async_generator_path" ]
+	then
+		return
+	fi
+
+	cd Tools
+
+	if [ ! -f nuget.exe ]
+	then
+		wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+	fi
+
+	async_generator_path="CSharpAsyncGenerator.CommandLine.$(cat packages.config | grep id=\"CSharpAsyncGenerator.CommandLine | cut -d\" -f4)/tools"
```

Comment:
Hack for parsing the async generator version out of the packages.config file:
 - Output the content of packages.config
 - Filter out any line not containing `id=\"CSharpAsyncGenerator.CommandLine`
 - Split the remaining line(s) by `"` and take the fourth string resulting of the split

If you know a better way, of course propose it.
I need it to build the path to the async generator version we have to use.